### ISSUE TITLE
NullReferenceException mapping from interface

### ISSUE THREAD

**Issue by kwaclaw** [agileobjects/AgileMapper#11]
It seems mapping from an interface is not supported?
This test case results in a null reference exception:
```
        public class InterfaceTest: IInterfaceTest
        {
            public string Name { get; set; }
        }

        public interface IInterfaceTest
        {
            string Name { get; set; }
        }

        public class InterfaceTest2
        {
            public string Name { get; set; }
        }

        [Fact]
        public void MapInterface() {
            var objImpl = new InterfaceTest { Name = "Test" };
            IInterfaceTest obj = objImpl;
            var obj2 = Mapper.Map(obj).ToANew<InterfaceTest2>();
            //var obj2 = new DictTest2 { JobDataMap = new Dictionary<string, object>() };
            //Mapper.Map(obj).Over(obj2);
        }
```

**Comment by SteveWilkes** [comment]
I've committed [a fix](https://github.com/agileobjects/AgileMapper/commit/84cf4b291daa5414b514e00587f74df6d44686d0) for this - I'll sort out the other issues before doing another release.

**Comment by SteveWilkes** [comment]
Fixed in v0.10.1 - thanks for the bug report! :)
